essentials-framework/Essentials Core/PepperDashEssentialsBase/Crestron IO/DinIo8/DinIo8Controller.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using Crestron.SimplSharpPro;
+using Crestron.SimplSharpPro.DeviceSupport;
+using Crestron.SimplSharpPro.GeneralIO;
+using PepperDash.Core;
+using PepperDash.Essentials.Core.Bridges;
+using PepperDash.Essentials.Core.Config;
+
+namespace PepperDash.Essentials.Core.CrestronIO
+{
+    public class DinIo8Controller:CrestronGenericBaseDevice, IIOPorts
+    {
+        private DinIo8 _device;
+
+        public DinIo8Controller(string key, Func<DeviceConfig, DinIo8> preActivationFunc, DeviceConfig config):base(key, config.Name)
+        {
+            AddPreActivationAction(() =>
+            {
+                _device = preActivationFunc(config);
+
+                RegisterCrestronGenericBase(_device);
+            });
+        }
+
+        #region Implementation of IIOPorts
+
+        public CrestronCollection<Versiport> VersiPorts
+        {
+            get { return _device.VersiPorts; }
+        }
+
+        public int NumberOfVersiPorts
+        {
+            get { return _device.NumberOfVersiPorts; }
+        }
+
+        #endregion
+
+
+    }
+
+    public class DinIo8ControllerFactory : EssentialsDeviceFactory<DinIo8Controller>
+    {
+        public DinIo8ControllerFactory()
+        {
+            TypeNames = new List<string>() { "DinIo8" };
+        }
+
+        public override EssentialsDevice BuildDevice(DeviceConfig dc)
+        {
+            Debug.Console(1, "Factory Attempting to create new C2N-IO Device");
```

Comment:
Please change the log statement to reflect the correct device

Reply:
COmplete